### ISSUE TITLE
issue node v0.12.3 win7 : problem processing multiple requests in parallel with cluster module (1 connection out of 2)

### ISSUE THREAD

**Issue by kristofen** [joyent/node#25350]
Hello, 
I've been trying nodejs 0.12..3 on my windows 7 pc (4 cores).
My test is focused on multicore connection processing (see the code below) using cluster module.

```javascript
var cluster = require('cluster');
var http = require('http');
var numCPUs = require('os').cpus().length;

if (cluster.isMaster) {
  // Fork workers.
  for (var i = 0; i < numCPUs; i++) {
    cluster.fork();
  }

  cluster.on('exit', function (worker, code, signal) {
    console.log('worker ' + worker.process.pid + ' died');
  });
} else {
  myApp();
}

function myApp() {
  // Workers can share any TCP connection
  // In this case its a HTTP server
  http.createServer(function (req, res) {
    console.log('[httpServer] new request');
    res.writeHead(200);
    // connection will be hanging forever 
    while (true) {
      for (var i = 0; i < 10000000; i++) {}
      res.write("chunk!!!");
    }
    res.end("this is the end\n");
  }).listen(3000, function (err) {
    if (err) console.error(err);
    console.log('listening on port 3000');
  });
}
```

My test is : launch 8 connections in parallel (with curl). Note: I launch 8 connections because I'm  on a 4 core PC.
The result: only connections number 1, 3, 5 and 7 will be processed  in parallel (ie: receiving chunk!chunk!chunk!...)

![sans titre](https://cloud.githubusercontent.com/assets/4066470/7699417/14f8ecd6-fe18-11e4-82a8-48ccfa3a160f.png)

**Comment by cjihrig** [comment]
What are you expecting to happen? You have four child processes. Each one receives a connection and then you put it into an infinite loop so that it can't accept any new connections.

**Comment by kristofen** [comment]
Thx for your reply

I was expecting that the first 4 connections will be processed and any additional connection will be on hold.
But the test shows that: 
- the first conn. is processed 
- the second is on hold
- the third is processed 
- the fourth is on hold 
- the fifth is processed 
- the sixth is on hold 
- the seventh is processed
- the eighth is on hold

Isn't that weird?

I was expecting 
- the first conn. is processed 
- the second is processed 
- the third is processed 
- the fourth is processed 
- the fifth is on hold (until one previous ends)
- the sixth is on hold  (until one previous ends)
- the seventh is on hold  (until one previous ends)
- the eighth is on hold (until one previous ends)

Am I missing something ?

**Comment by cjihrig** [comment]
I believe that on Windows, the operating system controls which requests are sent to which threads. I wonder if you'd have different results on a different OS using the round robin scheduler.

**Comment by kristofen** [comment]
thanks I'll test that on debian

**Comment by kristofen** [comment]
You're totally right @cjihrig !! on debian the first 4 connections are processed while any other connections are on hold.

Windows :-1:
